### PR TITLE
Extended cosine schedule T_max=60 on SOTA base config (50-epoch run)

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

Extending the cosine annealing period beyond the training window (T_max=60 on a 50-epoch run) gives a longer, gentler LR tail. The model sees more of its training at intermediate LR values rather than converging rapidly to near-zero, which should improve final generalization — especially for volume pressure which benefits from sustained gradient signal.

**Reference evidence:**
- Pre-wave run `5o7jc7wi` (T_max=13 on ~13-epoch run) achieved test_agg=8.313%, vol=11.867% — best volume score in the pre-wave reference table. That experiment was censored (timed out) and used an older base config.
- The SOTA run `sogus8sx` (T_max=50, i.e. default = epochs) achieved test_agg=7.9303%.
- T_max=60 on a 50-epoch run means the scheduler reaches ~cos(50/60 × π) ≈ 0.259×lr_range at final epoch — LR decays to only ~26% of peak rather than going to lr_min. This is meaningfully different from T_max=50.

**Why T_max=60 specifically:** Setting T_max slightly beyond the run length (1.2× epochs) is a standard trick that prevents the LR from hitting lr_min too early. The cosine schedule with T_max=60 and lr_min=0 on a 50-epoch run means:
- At epoch 25 (midpoint): LR ≈ 50% of peak
- At epoch 50 (end): LR ≈ 26% of peak (vs 0% with T_max=50)

This gives the optimizer sustained signal through the final epochs instead of a near-zero LR that stops learning.

**No code changes required.** This is a pure CLI experiment using the existing `--lr-cosine-t-max` flag.

## Instructions

This is a **pure CLI experiment** — no code changes to `train.py` or any source file. The only change from the SOTA base config is `--lr-cosine-t-max 60` instead of the default (which equals epochs=50).

### Smoke Run (DDP8, 3 epochs)

Confirm training is stable before committing to 50 epochs:

```bash
torchrun --nproc_per_node=8 train.py \
  --agent tanjiro \
  --wandb-group extended-cosine-t60-sota \
  --epochs 3 \
  --batch-size 2 \
  --lr 1e-4 \
  --optimizer lion \
  --use-ema \
  --ema-decay 0.999 \
  --lr-warmup-epochs 1 \
  --lr-cosine-t-max 60 \
  --train-surface-points 40000 \
  --eval-surface-points 40000 \
  --surface-loss-weight 1.0 \
  --volume-loss-weight 1.0 \
  --model-pe string_multisigma \
  --pe-init-sigmas "0.25,0.5,1.0,2.0,4.0" \
  --no-compile-model \
  --kill-thresholds "5494:val_primary/abupt_axis_mean_rel_l2_pct>25"
```

**Smoke pass:** EP1 val `abupt_axis_mean_rel_l2_pct` < 25%. This smoke should be essentially identical to SOTA run `sogus8sx` at EP1 since T_max only affects later LR decay.

### Long Run (DDP8, 50 epochs)

If smoke passes:

```bash
torchrun --nproc_per_node=8 train.py \
  --agent tanjiro \
  --wandb-group extended-cosine-t60-sota \
  --epochs 50 \
  --batch-size 2 \
  --lr 1e-4 \
  --optimizer lion \
  --use-ema \
  --ema-decay 0.999 \
  --lr-warmup-epochs 1 \
  --lr-cosine-t-max 60 \
  --train-surface-points 40000 \
  --eval-surface-points 40000 \
  --surface-loss-weight 1.0 \
  --volume-loss-weight 1.0 \
  --model-pe string_multisigma \
  --pe-init-sigmas "0.25,0.5,1.0,2.0,4.0" \
  --no-compile-model \
  --kill-thresholds "5494:val_primary/abupt_axis_mean_rel_l2_pct>20,10988:val_primary/abupt_axis_mean_rel_l2_pct>15,21976:val_primary/abupt_axis_mean_rel_l2_pct>10,32964:val_primary/abupt_axis_mean_rel_l2_pct>8"
```

**Kill gates (long run):**
- EP1 (~5,494 steps): abupt > 20% → kill
- EP2 (~10,988 steps): abupt > 15% → kill
- EP4 (~21,976 steps): abupt > 10% → kill
- EP6 (~32,964 steps): abupt > 8% → kill (must be on trajectory to beat wave SOTA val=6.5281%)

**Note:** Steps per epoch ≈ 5,494 at bs=2 with DDP8.

## Baseline to Beat

Current wave SOTA: **PR #599** (`sogus8sx`), test `abupt_axis_mean_rel_l2_pct` = **7.9303%**
- val best = **6.5281%**
- Config: Lion, lr=1e-4, bs=2, train_surface_points=40k, STRING multi-sigma PE (sigmas=[0.25,0.5,1.0,2.0,4.0]), no-compile-model, surface_loss_weight=1.0, **T_max=50 (default = epochs)**

Pre-wave reference `5o7jc7wi` (extended cosine T_max=13, **older config** — AdamW, smaller run):
- test_agg = 8.313%, volume = 11.867% (best volume in pre-wave table)

AB-UPT public targets (lower is better):
- surface_pressure_rel_l2_pct: 3.82
- wall_shear_rel_l2_pct: 7.29
- volume_pressure_rel_l2_pct: 6.08
- tau_x/tau_y/tau_z: 5.35 / 3.65 / 3.63

## Terminal Result Format

When complete, post a `SENPAI-RESULT` comment:

```
SENPAI-RESULT: {"terminal":true,"status":"complete","pending_arms":false,"wandb_run_ids":["<run-id>"],"primary_metric":{"name":"val_primary/abupt_axis_mean_rel_l2_pct","value":<number>},"test_metric":{"name":"test_primary/abupt_axis_mean_rel_l2_pct","value":<number>}}
```

Report val best checkpoint metrics (not terminal epoch) and full test metrics breakdown (surface, volume, wall shear components).

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: This PR has been open since it was assigned but has no experiment activity yet. Please begin the smoke run immediately.

This is a **pure CLI experiment** — no code changes required. Just run:

```bash
torchrun --nproc_per_node=8 train.py \
  --agent tanjiro \
  --wandb-group extended-cosine-t60-sota \
  --epochs 3 \
  --batch-size 2 \
  --lr 1e-4 \
  --optimizer lion \
  --use-ema \
  --ema-decay 0.999 \
  --lr-warmup-epochs 1 \
  --lr-cosine-t-max 60 \
  --train-surface-points 40000 \
  --eval-surface-points 40000 \
  --surface-loss-weight 1.0 \
  --volume-loss-weight 1.0 \
  --model-pe string_multisigma \
  --pe-init-sigmas '0.25,0.5,1.0,2.0,4.0' \
  --no-compile-model \
  --kill-thresholds '5494:val_primary/abupt_axis_mean_rel_l2_pct>25'
```

**Smoke pass criterion:** EP1 val `abupt_axis_mean_rel_l2_pct` < 25%. Since this is purely a cosine schedule change (T_max affects the LR tail, not EP1), the EP1 val should be essentially identical to the SOTA run `sogus8sx`.

Post the W&B run ID and EP1 val immediately when it logs. The only change from the SOTA config is `--lr-cosine-t-max 60`.

**Comment by morganmcg1** [comment]
ADVISOR: Second nudge — no response to the previous comment yet. 

**tanjiro, your GPU is idle.** This experiment requires zero code changes — it is a pure CLI flag experiment. You just need to run the command.

Launch the 3-epoch smoke run immediately:
```bash
torchrun --nproc_per_node=8 train.py \
  --agent tanjiro \
  --wandb-group extended-cosine-t60-sota \
  --epochs 3 \
  --batch-size 2 \
  --lr 1e-4 \
  --optimizer lion \
  --use-ema \
  --ema-decay 0.999 \
  --lr-warmup-epochs 1 \
  --lr-cosine-t-max 60 \
  --train-surface-points 40000 \
  --eval-surface-points 40000 \
  --surface-loss-weight 1.0 \
  --volume-loss-weight 1.0 \
  --model-pe string_multisigma \
  --pe-init-sigmas "0.25,0.5,1.0,2.0,4.0" \
  --no-compile-model \
  --kill-thresholds "5494:val_primary/abupt_axis_mean_rel_l2_pct>25"
```

Post EP1 val metrics as soon as they are available. Smoke pass criterion: EP1 val_abupt < 25%.

**Comment by morganmcg1** [comment]
STUDENT tanjiro: Smoke run launched after fixing missing `lion-pytorch` dependency (`pip install lion-pytorch` — already listed in pyproject.toml but not installed in pod env). Smoke run is now training.

- W&B (rank0): https://wandb.ai/wandb-applied-ai-team/senpai-v1-drivaerml-ddp8/runs/rse0oh3w
- W&B group: `extended-cosine-t60-sota`
- Config: 3 epochs, DDP8, --lr-cosine-t-max 60 (only deviation from `sogus8sx` SOTA)
- Will report EP1 val_abupt as soon as it logs.
